src/Poseidon/GenotypeData.hs

```diff
@@ -60,7 +60,11 @@ instance FromJSON GenotypeDataSpec where
         <*> v .:? "snpFileChkSum"
         <*> v .: "indFile"
         <*> v .:? "indFileChkSum"
-        <*> v .: "snpSet"
+        <*> (convertMaybeSnpSet <$> (v .:? "snpSet"))
+      where
+        convertMaybeSnpSet :: Maybe SNPSetSpec -> SNPSetSpec
+        convertMaybeSnpSet Nothing = SNPSetOther
+        convertMaybeSnpSet (Just s) = s
```

Comment:
It certainly feels a bit hacky, but it's pragmatic :shrug:

Reply:
Oh no it's not hacky at all! On the JSON/YAML file, this is now formally a non-mandatory field, as described in the Schema, but on the internal data structure side, this will be a non-Maybe field. I think this is a nice and clean handler that mediates between these two worlds :-)

Reply:
Alright, alright :smile: - just generally the two worlds should be as close to each other as possible.

Reply:
Yes, that's true, and it's certainly a design choice that one can debate. An alternative would be to fully reflect to the non-mandatory nature also in our data structure and just _interpret_ it like `SnpSetOther` whenever it needs to, but that I also find a bit cumbersome.

Reply:
Hmm... actually now that I said it, I think that's better... perhaps I should make that change and see how it looks. Once cool advantage is that it would allow us to see, for example in `forge` whether there are package with `Nothing` in snpSet and make the warning even more explicit. I'll give it a go.